backend/matching_service/controllers/socketController.js

```diff
@@ -153,9 +190,30 @@ class SocketController {
     }
   }
 
-  async handleDisconnect(socket) {
-    await this.removeExistingConnection(socket.handshake.auth.uid); // uid is passed in socket auth from frontend
+  emitNavigateToCollab(prevUserSocket, currUserSocket, sessionId, prevUserSessionData, currUserSessionData, questionData) {
+    console.log("Emitting navigateToCollab with params:", {
+      sessionId,
+      prevUserSessionData,
+      currUserSessionData,
+      questionData
+    });
+    // Navigate to collab and pass session info
+    prevUserSocket.emit("navigateToCollab", { sessionId, sessionData: prevUserSessionData, questionData });
+    currUserSocket.emit("navigateToCollab", { sessionId, sessionData: currUserSessionData, questionData });
+    console.log("Event 'navigateToCollab' emitted to both sockets.");
   }
+  
+  async handleDisconnect(socket) {
```

Comment:
Perhaps it would be better to make `handleDisconnect` be purely a single connection disconnection function, rather than it having any functionality to do with `otherUID`. This is because matching service itself (which has no concept of `otherUID`), does use this function as well.

In addition, relating to a comment I made in `frontend/MatchingButton.tsx`, the otherUID field is not set in the frontend, so this would be a empty field anyways. 

Making `handleDisconnect` to only be concerned about a single connection would still work for our collab use case, as long as we perform `.disconnect()` in `frontend/collabPageView` upon `sessionTerminated`.